source/MaterialXGenShader/Util.cpp

```diff
@@ -313,6 +313,29 @@ namespace
                         ++numOpaque;
                     }
                 }
+                else
+                {
+                    NodeDefPtr nodeDef = node->getNodeDef();
+                    const string& typeName2 = nodeDef->getType();
```

Comment:
Can't say I love the naming convention of typeName and typeName2 or type/type2, etc. To me it doesn't make it clear what the purpose of each variable is. If I understand correctly, typeName is really the upstreamNodeTypeName and typeName2 is the nodeTypeName. Maybe some convention like this would add clarity to the code.